crates/layout/tests/layout_tests.rs

```diff
@@ -235,6 +235,127 @@ fn test_replacement_fixedlayout() {
     clear_views();
 }
 
+#[test]
+fn test_vertical_fill() {
+    let figma_doc_result = setup();
+    assert!(figma_doc_result.is_ok());
+
+    let figma_doc = figma_doc_result.unwrap();
+    let view_result = figma_doc.views.get(&NodeQuery::NodeName("VerticalFill".into()));
+    assert!(view_result.is_some());
+
+    let view = view_result.unwrap();
+    let mut id = 0;
+    add_view_to_layout(&view, &mut id, -1, -1, &HashMap::new(), &figma_doc.views);
+
+    let root_layout_result = get_node_layout(0);
+    assert!(root_layout_result.is_some());
+    let root_layout = root_layout_result.unwrap();
+
+    assert!(root_layout.width == 150.0);
+    assert!(root_layout.height == 130.0);
+
+    // Right node should fill to fit root
+    let right_layout_result = get_node_layout(2);
+    assert!(right_layout_result.is_some());
+    let right_layout = right_layout_result.unwrap();
+    assert!(right_layout.width == 70.0);
+    assert!(right_layout.height == 110.0);
+    assert!(right_layout.left == 70.0);
+    assert!(right_layout.top == 10.0);
+
+    // Auto fill height node should fill to fit Right
+    let auto_fill_height_result = get_node_layout(3);
+    assert!(auto_fill_height_result.is_some());
+    let auto_fill_height = auto_fill_height_result.unwrap();
+    assert!(auto_fill_height.width == 70.0);
+    assert!(auto_fill_height.height == 30.0);
+
+    // Increase fixed left node height by 30 pixels
+    let result = set_node_size(1, 50, 140);
+    assert!(result.changed_layout_ids.contains(&2));
+    assert!(result.changed_layout_ids.contains(&3));
+
+    // Right node should be taller by 30 pixels
+    let right_layout_result = get_node_layout(2);
+    assert!(right_layout_result.is_some());
+    let right_layout = right_layout_result.unwrap();
+    assert!(right_layout.width == 70.0);
+    assert!(right_layout.height == 140.0);
+    assert!(right_layout.left == 70.0);
+    assert!(right_layout.top == 10.0);
+
+    // Auto fill height node should be taller by 30 pixels
+    let auto_fill_height_result = get_node_layout(3);
+    assert!(auto_fill_height_result.is_some());
+    let auto_fill_height = auto_fill_height_result.unwrap();
+    assert!(auto_fill_height.width == 70.0);
+    assert!(auto_fill_height.height == 60.0);
+
+    clear_views();
+}
+
+#[test]
+fn test_horizontal_fill() {
```

Comment:
Could you split these into separate tests?
go/unit-testing-overview#focused

Reply:
Same for the Vertical tests

Reply:
You mean split the part where it gets resized into a separate test?

Reply:
Well, like this for instance:
```
    // Increase fixed left node height by 30 pixels
    let result = set_node_size(1, 50, 140);
    assert!(result.changed_layout_ids.contains(&2));
    assert!(result.changed_layout_ids.contains(&3));
```
That looks like a pretty independent bit, does it rely on everything that came before it in the test?

Reply:
It relies on the first 10 lines or so. I guess I can move the second part of this test out to a second test, but it doesn't seem very beneficial to me.

Reply:
But the whole test can be split into independent tests. I just tried it and split out the first three chunks of the test into their own functions: https://gist.github.com/timothyfroehlich/434f0c7b890a3f03a43354ffb1b56625

Here, this goes into more detail about why we should be writing smaller, more focused tests: go/unit-testing-practices#behavior-testing

Reply:
Separated into two tests for vertical fill and two for horizontal

Reply:
I'd still prefer them to be completely split into individual test cases, but this is better. I'm bringing this up because I'm having to clean up some of my own tests in #459 because I had too many unrelated things in the same test that didn't need to be there.

Reply:
Added a comment in https://github.com/google/automotive-design-compose/issues/405